spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -2587,13 +2587,21 @@ void ParserBuilder::skip(Expression* size, const Meta& location) {
 void ParserBuilder::advanceToNextData() {
     auto old_offset = builder()->addTmp("old_offset", builder()->memberCall(state().cur, "offset"));
 
+    // Call any `%sync_advance` hook as we enter sync mode.
+    guardFeatureCode(state().unit, {"uses_sync_advance"}, [&]() {
+        beforeHook();
+        builder()->addMemberCall(state().self, "__on_0x25_sync_advance", {old_offset});
+        afterHook();
+    });
+
```

Comment:
Is this really the right location for this code? `advanceToNextData()` is called from multiple places and I don't think all them are indicating the beginning of sync mode. With this code here, I think we essentially just run the hooks twice each time, once before we actually move `cur` and once before.

Reply:
Yes, this is currently not the right place. Let me look again (while avoiding calling it at the same offset multiple times).

Reply:
Resolving this for now, opened a new issue for calling the hook multiple times at the same offset.